sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/UniversalControlFlowGraphBuilder.cs

```diff
@@ -84,25 +84,22 @@ private BasicBlock CreateBasicBlock(Block block)
             {
                 instructionBuilder.BuildInstruction(jump.JumpNode);
             }
-            else
+
+            if (block is ExitBlock exit)
             {
-                if (block.SuccessorBlocks.All(cfg.ExitBlock.Equals))
-                {
-                    basicBlock.Ret = new Return { ReturnedExpression = ConstantExpression };
-                }
-                else
-                {
-                    basicBlock.Jump = new Jump();
-                    basicBlock.Jump.Destinations.AddRange(block.SuccessorBlocks.Select(blockId.Get));
-                }
+                basicBlock.Ret = new Return { ReturnedExpression = ConstantExpression };
```

Comment:
ExitBlock always returns constant. If the method is not void, the corresponding returns will be handled by line 85